libp2p/stream/chronosstream.nim

```diff
@@ -65,11 +68,27 @@ template withExceptions(body: untyped) =
     # TODO https://github.com/status-im/nim-chronos/pull/99
     raise newLPStreamEOFError()
 
+proc trackPeerIdentity(s: ChronosStream) =
+  if not s.tracked:
+    if not isNil(s.peerInfo) and s.peerInfo.agentVersion.len > 0:
+      let name = s.peerInfo.agentVersion.split(" ")[0]
+      libp2p_peers_identity.inc(labelValues = [name])
+      s.tracked = true
+
+proc untrackPeerIdentity(s: ChronosStream) =
+  if s.tracked:
+    if not isNil(s.peerInfo) and s.peerInfo.agentVersion.len > 0:
+      let name = s.peerInfo.agentVersion.split(" ")[0]
+      libp2p_peers_identity.dec(labelValues = [name])
+      s.tracked = false
+
 method readOnce*(s: ChronosStream, pbytes: pointer, nbytes: int): Future[int] {.async.} =
   if s.atEof:
     raise newLPStreamEOFError()
 
   withExceptions:
+    s.trackPeerIdentity()
```

Comment:
we need to track to amount of bytes received.

Reply:
this should also be moved after the read, same for write, should only trigger after the write happened.

Reply:
good point